config/routes.rb

```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users#, only: [:new, :create, :edit, :show, :index]
+  root 'users#new'
 end
```

Comment:
You should have users' resources restricted to only `resources :users, only: [:new, :create, :edit, :update]`. Also, it seems you have commented out these restrictions kindly fix that also.